rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -430,6 +432,42 @@ def run_all_post_decoding_rules(
 
         return decoded_events
 
+    def _is_spam_airdrop(
+            self,
+            transaction: EvmTransaction,
+            tx_receipt: EvmTxReceipt,
+    ) -> bool:
+        """
+        Detect spam transactions that can be heavy to process by the decoding logic.
+
+        We focus on airdrops where we have a huge number of log events all of them transfers.
+        First check if we are above the SPAM threshold, then we check if all the events are
+        transfers (we do it in sqlite for better performance) and lastly we check if the
+        transferred token is marked as spam. For example the OP airdrops have such transactions
+        and we do check the protocol to avoid false positives.
+        """
+        if len(tx_receipt.logs) < SPAM_LOG_EVENTS_THRESHOLD:
+            return False
+
+        with self.database.conn.read_ctx() as cursor:
+            tx_id = transaction.get_or_query_db_id(cursor)
+            cursor.execute(
+                'SELECT DISTINCT topic, address FROM evmtx_receipt_logs JOIN '
+                'evmtx_receipt_log_topics ON evmtx_receipt_logs.identifier = '
+                'evmtx_receipt_log_topics.log WHERE evmtx_receipt_logs.tx_id=? AND topic_index=0',
+                (tx_id,),
+            )
+            data = cursor.fetchall()
+            if {entry[0] for entry in data} == set((ERC20_OR_ERC721_TRANSFER,)):  # noqa: C405  # we don't want a set of the chars but a set with one element
```

Comment:
can't you do this check in sqlite itself?
Also if you wanted a set with one element just do `{((ERC20_OR_ERC721_TRANSFER,)}`